### ISSUE TITLE
Input data plots

### ISSUE THREAD

**Issue by MikkoVihtakari** [gadget-framework/gadgetplots#2]
I think it would be nice to have a possibility to plot input data to see how they look before passing them into a gadget model. I have some functions in my ghl model project which I could add. Now the question is how to name them to separate them from model output figures?

Output figures are currently named as `plot_*`. Would `dplot_` (data plot) work? Should output figures then be renamed `mplot_*` or should we keep them as they are for consistency with the rgadget package?

**Comment by bthe** [comment]
That sounds like an excellent idea.

@lentinj and I have discussed in the past the need to for a data explorer on top of MFDB, as you would want explore many different datasets before you make the final decision of what to include (it was actually the first issue on: https://github.com/gadget-framework/mfdb/issues/1 ). 

 Wrt. naming, how about `g3_plot_*` and `mfdb_plot_*`?

**Comment by MikkoVihtakari** [comment]
I am also thinking about making it possible to not use mfdb with gadget3. Just writing a function which I will send your way when I have a draft ready. Hence `mfdb_plot_*` might not be the best name...`gplot_*` and `dplot_*` would allow the quick use of tabulator for code completion to get the alternatives.

**Comment by MikkoVihtakari** [comment]
Thanks for the clarification @lentinj, I did not quite get it :P Could you elaborate a bit? If I had R model in an object/function called `model`, how would I extract the datta?

**Comment by lentinj** [comment]
With a model setup like:-

```
> prey_a <- g3_stock('prey_a', seq(1, 5, by = 1)) %>% g3s_age(1,5) %>% g3s_livesonareas(1:2)
> prey_a__init <- g3_stock_instance(prey_a)
> prey_a__init[] <- runif(length(prey_a__init))
> obsdata <- expand.grid(
+     year = 2000,
+     length = seq(1, 5, by = 1),
+     age = 1:5)
> obsdata$number <- runif(nrow(obsdata))
> model_fn <- g3_to_r(list(
+     g3a_time(2000, 2001),
+     g3a_initialconditions(prey_a,
+         num_f = g3_formula(stock_ss(prey_a__init), prey_a__init = prey_a__init),
+         wgt_f = 10),
+     g3l_abundancedistribution("nt",
+         obsdata,
+         function_f = g3l_distribution_sumofsquares(),
+         stocks = list(prey_a),
+         report = TRUE)))
```

You can list variables stored in the function's environment:

```
> ls(environment(model_fn))
[1] "adist_sumofsquares_nt_obs__num" "prey_a__areas"
[3] "prey_a__init"
```

...and pull out those variables:

```
> environment(model_fn)$adist_sumofsquares_nt_obs__num
, , time = 2000

       age
length       age1      age2       age3      age4       age5
  1:2   0.7892528 0.5289322 0.10441556 0.7476979 0.39173710
  2:3   0.9126536 0.5006091 0.96345715 0.9044553 0.83099030
  3:4   0.7276539 0.0640001 0.92904490 0.3630014 0.64497698
  4:5   0.4347278 0.8150522 0.06269184 0.6488409 0.22146712
  5:Inf 0.6613986 0.6324099 0.54975972 0.6790018 0.01533964
```

**Comment by MikkoVihtakari** [comment]
Thanks Jamie. I started. Need to do other things now. Back in gadget business in 2023.

**Comment by MikkoVihtakari** [comment]
I added a function to plot all the data in a model now: https://github.com/gadget-framework/gadgetplots/commit/34ab2c21c31e64a1a0604c8ad892263cdb796a21

The function is called `g3d_plot`. It's only tested for NEA Ghl model and almost certainly won't work for all possible model configurations. I had to take many assumptions. Anyway, this is a start. Will add my data plot functions later this week.
